pennylane/ops/qubit.py

```diff
@@ -2505,6 +2505,92 @@ def diagonalizing_gates(self):
         """
         return [QubitUnitary(self.eigendecomposition["eigvec"].conj().T, wires=list(self.wires))]
 
+# =============================================================================
+# Arithmetic
+# =============================================================================
+
+class Carry(Operation):
+    r"""Carry()
+    Apply a ``Carry`` operation on input wires
+    TODO add more information
+
+    **Details:**
+
+    * Number of wires: 4
+    * Number of parameters: 0
+    TODO consider gradient
+    * Gradient recipe: TODO
+
+    Args:
+        wires (Union[Wires or int]): the wires the operation acts on
+
+    **Example**
+    TODO
+    """
+    num_params = 0
+    num_wires = 4
+    par_domain = None
+    matrix = np.array([[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]])
+
+    @classmethod
+    def _matrix(self, *params):
+        return Carry.matrix
+    
+    def adjoint(self, do_queue=False):
+        return Carry(wires=self.wires,do_queue=do_queue)
```

Comment:
Nice! Maybe you're already working on it, but adding the `expand()` method will be crucial